neptune/new/internal/operation_processors/async_operation_processor.py

```diff
@@ -189,6 +203,14 @@ def __init__(
             self._batch_size = batch_size
             self._last_flush = 0
 
+        def run(self):
+            try:
+                super().run()
+            except Exception:
+                with self._processor._waiting_cond:
```

Comment:
I guess it's the part responsible for faster dying when sync process is already dead, but I don't know how.
Some commet would be appreciated ;)

Reply:
It just wakes waiting threads when synchronization process is dying. If not for this they would be sleeping forever waiting for synchronization. Nothing magic here.